report_viewer/apps/viewer/tests.py

```diff
@@ -169,3 +174,65 @@ def test_view_report_not_on_s3(client):
     assert response.status_code == 200
 
     assertContains(response, html_on_db)
+
+
+def test_report_metric_middleware_client_ip():
+    get_response = mock.MagicMock()
+    factory = RequestFactory()
```

Comment:
`django-pylint` has a built in request factory fixture (called `rf`). See https://pytest-django.readthedocs.io/en/latest/helpers.html#rf-requestfactory

Reply:
Nice. Added.